### ISSUE TITLE
Client-side word cloud?

### ISSUE THREAD

**Issue by rloomans** [zpurcey/new-teamtemp#14]
@zpurcey are there any reasons not to replace the word cloud with a client-side library like https://github.com/timdream/wordcloud2.js?

**Comment by zpurcey** [comment]
I recall testing with every js implementation I could find back in the day. Would have been my preference.

The biggest tricks were ensuring the algorithm would reliably display ALL words and zoom appropriately to fit them all.

Couple of good tests if you wanted to try a client:
- Will it support 1 long word amoung a few regular sized words (supercolorfragilisticexpealidious)
- Display a word cloud in the same size box when you have 20 words or 500 and still not drop anyone's word because people look to see 'their' word.
- Always display all words. Don't drop words like and or but

As I stepped through a few different options I'd run into road blocks due to the library flexibility. In the end I built the generator by hand using r so I'd have complete control.

Are you seeing issues with the current generator?

**Comment by zpurcey** [comment]
Interested to hear what you think. Had a look at wordcloud2.js and don't see any obvious reasons why it wouldn't work just playing with the demo.

**Comment by rloomans** [comment]
The principle problem I had was spinning up a new instance to experiment on: without the API key, the word cloud wouldn't turn up. 

In the near future I want to install the app inside our corporate environment and even with the API key, it doesn't make much sense to be still calling out to an external service for the word cloud. 

## 

Robert Loomans

**Comment by zpurcey** [comment]
Yeah understand. I suggest that the js client should do it with a little
work to wrap it. I like it's multi language support and the ability to form
the word cloud as a shape. Let me know how you go or reach out if you just
want to internalize the r service.

Cheers,

Andrew
